sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -92,6 +99,14 @@ abstract class QueryGenerator(private val query: BindableQuery) {
     // A list of arrays we've encountered so far.
     val precedingArrays = mutableListOf<String>()
 
+    val extractedVariables = mutableMapOf<IntermediateType, String>()
+    // extract the variable for duplicate types, so we don't encode twice
+    for (type in duplicateTypes) {
+      val encodedJavaType = type.encodedJavaType() ?: continue
+      val variableName = "encoded${type.nameCapitalized}"
```

Comment:
This is not sufficient mangling to avoid a conflict with another parameter that may have this name.

We need to create a `NameAllocator`, populate it first with all of the parameter names, and then ask it to create a new name for the local variable using the parameter name (which will usually suffix an underscore to avoid collision with the actual parameter name).

Let me know if you want me to describe that more or show code on how to accomplish it.

Reply:
thanks! no worries, I saw its use just below and it's already prepopulated with all the type names